frontend/amundsen_application/static/js/config/config-types.ts

```diff
@@ -151,7 +151,7 @@ export enum NoticeSeverity {
 }
 export interface NoticeType {
   severity: NoticeSeverity;
-  messageHtml: string;
+  messageHtml: string | ((resourceName) => string);
```

Comment:
I think this would work better:
```suggestion
  messageHtml: string | (string) => string;
```